src/Nox.Solution/Providers/EnvironmentVariableValueProvider.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Nox.Solution.Utils;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Nox.Solution.Macros;
+
+public class EnvironmentVariableValueProvider
+
+{
+    private readonly IEnvironmentProvider _environmentProvider;
+
+    public EnvironmentVariableValueProvider(IEnvironmentProvider environmentProvider)
+    {
+        _environmentProvider = environmentProvider;
+    }
+
+    public IReadOnlyDictionary<string, string?> Resolve(
+        IReadOnlyList<string> variables, 
+        IReadOnlyDictionary<string,object>? defaults = null)
+    {
+        var values = new Dictionary<string, string?> ();
```

Comment:
for better performance
```suggestion
        var values = new Dictionary<string, string?> (variables.Count);
```